packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -2424,7 +2424,7 @@ export class Bucket extends BucketBase {
     return this.inventories.map((inventory, index) => {
       const format = inventory.format ?? InventoryFormat.CSV;
       const frequency = inventory.frequency ?? InventoryFrequency.WEEKLY;
-      const id = inventory.inventoryId ?? `${this.node.id}Inventory${index}`;
+      const id = inventory.inventoryId ?? `${this.node.id}Inventory${index}`.replace( /[^\w\.\-]/g, '').slice(-64);
```

Comment:
If `inventoryId` is explicitly provided with bad characters / is too long, I think we should validate with a clear error message.

Can you also put an explanation in the doc string (~ line 1253).

I can understand replacing bad characters if they are taken from the node id.

Reply:
Done